blueprints/test-framework-detector.js

```diff
@@ -12,7 +14,12 @@ module.exports = function(blueprint) {
     if ('ember-cli-qunit' in dependencies) {
       type = 'qunit';
     } else if ('ember-cli-mocha' in dependencies) {
-      type = 'mocha';
+      if (fs.existsSync(this.path + '/mocha-0.12-files') &&
+        semver.satisfies('0.12.0', dependencies['ember-cli-mocha'])) {
+        type = 'mocha-0.12';
+      } else {
+        type = 'mocha';
```

Comment:
I think we need to flip-flop this logic, I (and I could be going out on a limb here) think that the new syntax will likely stay around even if we have to make a 0.13 / 1.0 / etc of ember-mocha.  We should update this conditional so the `else` is the new format.

I'd also suggest `mocha-current` and `mocha-lt-0.12` as the folder paths.

Thoughts?